compliance/cloud_credentials/aws/aws_connection_key_rotation_policy.pt

```diff
@@ -0,0 +1,323 @@
+# AWS Credential Update Policy
+#
+# This policy updates the credentials used to connect RightScale to an AWS account.
+#
+
+name "AWS Key Rotation Policy"
+rs_pt_ver 20180301
+type "policy"
+short_description "Updates the IAM user keys used to connect RightScale to an AWS account."
+long_description "Version: 1.0"
+severity "medium"
+category "Compliance"
+
+permission "general_permissions" do
+  resources "rs_cm.credentials", "rs_cm.audit_entries", "rs_cm.clouds"
+  actions   "rs_cm.*"
+end
+
+auth "auth_rs", type: "rightscale"
+
+parameter "param_email" do
+  type "list"
+  label "List of emails address(es) to receive reports."
+end
+
+parameter "param_rotation_period" do
+  type "number"
+  label "Number of days between key updates."
+  default 90
+end
+
+# We use a RightScale account tag to keep track of when to rotate the keys.
+# If it not found, the policy will rotate keys at that time and add the tag for the next rotation.
+datasource "ds_account_tags" do
+  request do
+    auth $auth_rs
+    verb "POST"
+    host rs_cm_host
+    path "/api/tags/by_resource"
+    header "X-Api-Version", "1.5"
+    body_field "resource_hrefs", [join(["/api/accounts/",rs_project_id])]
+  end
+end
+
+# Get attached AWS clouds
+resources "aws_clouds", type: "rs_cm.clouds" do
+  filter do
+    cloud_type "amazon"
+  end
+end
+
+datasource "ds_aws_clouds" do
+  iterate @aws_clouds
+  field "href", href(iter_item)
+  field "name", val(iter_item,'name')
+end
+
+datasource "ds_cloud_accounts" do
+  request do
+    auth $auth_rs
+    host rs_cm_host
+    path "/api/cloud_accounts"
+    header "X-API-Version", "1.5"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[]") do
+      field "cloud_href", jmes_path(col_item, "links[?rel=='cloud'].href | [0]") 
+      field "cloud_account_href", jmes_path(col_item, "links[?rel=='self'].href | [0]")
+    end
+  end
+end
+
+datasource "ds_aws_cloud_accounts" do
+  run_script $js_aws_cloud_accounts, $ds_aws_clouds, $ds_cloud_accounts, $ds_account_tags, $param_rotation_period
+end
+
+# Find unattached volumes.
```

Comment:
There is a number of areas that volumes are mentioned.  should be cleaned up to reference clouds instead.